src/core/net/srp_server.cpp

```diff
@@ -693,14 +693,10 @@ Error Server::ProcessHostDescriptionInstruction(Host &                   aHost,
 
     VerifyOrExit(aHost.GetFullName() != nullptr, error = kErrorFailed);
     VerifyOrExit(aHost.GetKey() != nullptr, error = kErrorFailed);
-    {
-        uint8_t hostAddressesNum;
-
-        aHost.GetAddresses(hostAddressesNum);
 
-        // There MUST be at least one valid address if we have nonzero lease.
-        VerifyOrExit(aHost.GetLease() > 0 || hostAddressesNum > 0, error = kErrorFailed);
-    }
+    // We check the number of host addresses after processing of the
+    // Lease Option in the Addition Section and determining whether
+    // the host is being removed or registered.
```

Comment:
@abtink Thanks for catching this bug!

Reply:
Of course!  😃  I figured out quickly that we need to move it after "Lease Option" is processed (I copied the same block of code there) but then it still did not work.... it took me some time to notice that 
```VerifyOrExit(aHost.GetLease() > 0 || hostAddressesNum > 0, error = kErrorFailed);```
is not what we want and we really want 
```VerifyOrExit(aHost.GetLease() == 0 || hostAddressesNum > 0, error = kErrorFailed);```. 

When you read the `VerifyOrExit()` line it all seems correct 😃. I changed it to use `if()` instead which I think make it easier to read.

Reply:
`GetLease()` was always `0` in our previous code but we didn't detect this issue as all SRP messages always contain a valid address. 

> When you read the VerifyOrExit() line it all seems correct smiley. I changed it to use if() instead which I think make it easier to read.

Agree that it's easier to read!